src/guest_tracking.rs

```diff
@@ -96,18 +96,16 @@ impl<T: GuestStagePagingMode> GuestVm<T> {
         PageArc::<RwLock<GuestVmInner<T>>>::required_pages()
     }
 
-    /// Creates a new initializing `GuestVm` from `vm`, using `page` as storage.
-    pub fn new(vm: Vm<T>, page: Page<InternalClean>) -> Self {
-        // assert it fits in a single page for now.
-        assert!(Self::required_pages() == 1);
+    /// Creates a new initializing `GuestVm` from `vm`, using `pages`
+    /// as storage. Caller must ensure that `pages` contains
+    /// contiguous pages that can hold the PageArc.
+    pub fn new(vm: Vm<T>, pages: SequentialPages<InternalClean>) -> Self {
+        assert!(Self::required_pages() <= pages.len());
```

Comment:
Hmm. I can return the Sequential Pages back in the error, but somehow it gets dangerous because a simple '?' might resul t in leaked pages. I have become assert happy when handling SequentialPages.
I think the idea was to fail on the area of salus where we were passing around SequentialPages.

Reply:
I'll think about this in a next PR. I think if this change should change for all interface that now are dependent on SequentialPages.

Reply:
See `SequentialPages::from_pages`. It does something similar returning the passed pages back in the error case. Having the pages be part of the error type prevents them from being leaked, a `?` can't generate an error without the pages.

Reply:
Fair enough. So the strategy is having an error so annoying to deal with that you have to put unwrap, and explain why. It's not a bad one. :)